File: week3_Algorithms/struct.js
let print = console.log;
// const get_input = require('readline-sync');
// let answer = get_input.question("String: ");

let arrayOfPeople = [];

class Person {
  constructor(name, phone) {
    this.name = name;
    this.phone = phone;
  }
  addToArr() {
    arrayOfPeople.push({name: this.name, phone: this.phone});
    return this;
  }
}

let person1 = new Person("Holly", "0546553296").addToArr();
let person2 = new Person("David", "0523988545").addToArr();

const findName = "Holly";

function isExist() {
  for (let i = 0; i < arrayOfPeople.length; i++) {
    if (arrayOfPeople[i].name == findName) {
      print("Found:", arrayOfPeople[i].phone);
      return true;
    }
  }
  print("Not found");
  return false;
}

print(isExist());

File: week3_Algorithms/struct.c
#include <stdio.h>
#include <string.h>

typedef struct
{
  char *name;
  char *number;
}
person;

int main(void)
{
  char input[8] = {0};
  person people[2];

  people[0].name = "Holly";
  people[0].number = "0546553296";

  people[1].name = "David";
  people[1].number = "0523988545";

  printf("Name: ");
  scanf("%s", input);

  for (int i = 0; i < 5; i++)
  {
    if(strcmp(people[i].name, input) == 0)
    {
      printf("Found: %s\n", people[i].number);
      return 0;
    }
  }
  printf("Not found");
  return 1;  
}